ui/src/track-layout/track-layout-react-utils.tsx

```diff
@@ -202,30 +203,33 @@ export function useReferenceLineStartAndEnd(
 export function useLocationTrackStartAndEnd(
     id: LocationTrackId | undefined,
     publishType: PublishType | undefined,
-    ...changeTimes: TimeStamp[]
+    changeTimes: ChangeTimes,
 ): [AlignmentStartAndEnd | undefined, LoaderStatus] {
+    const changeTime = getMaxTimestamp(
+        changeTimes.layoutLocationTrack,
+        changeTimes.layoutTrackNumber,
+        changeTimes.layoutReferenceLine,
+    );
     return useLoaderWithStatus(
         () =>
             id && publishType
-                ? getLocationTrackStartAndEnd(
-                      id,
-                      publishType,
-                      getMaxTimestampFromArray(changeTimes),
-                  )
+                ? getLocationTrackStartAndEnd(id, publishType, changeTime)
                 : undefined,
-        [id, publishType, ...changeTimes],
+        [id, publishType, changeTime],
     );
 }
 
 export function useLocationTrackInfoboxExtras(
     id: LocationTrackId | undefined,
     publishType: PublishType,
-    locationTrackChangeTime: TimeStamp,
-    switchChangeTime: TimeStamp,
+    changeTimes: ChangeTimes,
 ): [LocationTrackInfoboxExtras | undefined, LoaderStatus] {
     return useLoaderWithStatus(
-        () => (id === undefined ? undefined : getLocationTrackInfoboxExtras(id, publishType)),
-        [id, publishType, locationTrackChangeTime, switchChangeTime],
+        () =>
+            id === undefined
+                ? undefined
+                : getLocationTrackInfoboxExtras(id, publishType, changeTimes),
+        [id, publishType, changeTimes],
```

Comment:
Tässä depsuna koko `changeTimes`-olio. En katsonut sen olevan niin paha juttu tässä, sillä tässä käytännössä vaan kutsutaan tuota ylempänä kommentoimaani hakua, jolla on kakun depsuina tarkemmat muutosajat. Saattaa ehkä aiheuttaa uudelleenpiirtoja, mutta ei sentään bäkkärihakuja. Erillisten muutosaikojen listailu täällä taas johtaisi uudelleen siihen, että meillä on kaksi erillistä muutosaikasettiä joista pitää pitää huolta.

Reply:
Joo hiukan ikävä kyllä molemmin päin. Varmaan ihan ok, ei nuo changetimet koko ajan vekslaile.